integration/kubernetes/k8s-number-cpus.bats

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../lib/common.bash"
+issue="https://github.com/kata-containers/runtime/issues/2186"
+
+setup() {
+	skip "test not working see: ${issue}"
+	export KUBECONFIG="$HOME/.kube/config"
+	pod_name="cpu-test"
+	container_name="c1"
+	get_pod_config_dir
+}
+
+@test "Check number of cpus" {
+	skip "test not working see: ${issue}"
+	# Create pod
+	kubectl create -f "${pod_config_dir}/pod-number-cpu.yaml"
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready pod "$pod_name"
+
+	retries="10"
+	ok="0"
+
+	for _ in $(seq 1 "$retries"); do
+		# Get number of cpus
+		max_number_cpus="3"
+		number_cpus=$(kubectl exec -ti pod/"$pod_name" -c "$container_name" nproc | sed 's/[[:space:]]//g')
+		# Verify number of cpus
+		if [ "$number_cpus" -le "$max_number_cpus" ]; then
+			ok="1"
+			break;
```

Comment:
you are checking the number of cpus just once,  if you remove the `skips` from test, it will pass when it shouldn't ,
instead you should exec `nproc` in the loop and verify its value. I think `ok` variable is not needed and `[ "$number_cpus" -le "$max_number_cpus" ]` should be enough